runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/ArtifactServiceStager.java

```diff
@@ -222,13 +230,26 @@ public StagingResult call() {
     }
   }
 
+  /** A file along with a staging name. */
+  @AutoValue
+  public abstract static class StagedFile {
```

Comment:
<!--thread_id:cc_182611630_t; commit:a16a978c74d440d81ca78b3c44be72eae312ef74; resolved:0-->
<!--section:context-quote-->
> **tgroh** wrote:
> We create this before staging, right? The name implies after

<!--section:body-->
See https://github.com/apache/beam/pull/5150#discussion_r182507804. I'm open to suggestions, but otherwise I'll leave it as is.